src/main/java/com/epam/controller/ToursController.java

```diff
@@ -98,30 +102,25 @@ public ModelAndView addTour(@RequestParam String tourType, @RequestParam String
     }
 
     @PostMapping("/updatetour")
-    public ModelAndView updateTour(@RequestParam String tourId, @RequestParam String tourType, @RequestParam String startDate, @RequestParam String endDate,
-                                   @RequestParam String country, @RequestParam String city, @RequestParam String hotel,
+    public ModelAndView updateTour(@RequestParam String tourId, @RequestParam String tourType,
                                    @RequestParam String pricePerPerson, @RequestParam String discount, @RequestParam String tourDescription) {
         ModelAndView toursModel = new ModelAndView();
-        toursModel.setViewName("updatetour");
         try {
-            LocalDate addStartDate = Validator.getDate(startDate, false);
-            LocalDate addEndDate = Validator.getDate(endDate, false);
+            Integer addTourId = Validator.getInt(tourId);
+            TourOffer tourOffer = toursOfferService.getTourById(addTourId);
+            toursModel.addObject("tour",tourOffer);
             Integer addPricePerPerson = Validator.getInt(pricePerPerson);
+            Integer addDiscount = Validator.getInt(discount);
             Validator.checkEmpty(tourType);
             Validator.checkEmpty(tourDescription);
-            Validator.checkDateDifferent(addStartDate, addEndDate);
-            int result = toursOfferService.updateTour(TourOffer.builder()
-                    .id(Integer.valueOf(tourId))
-                    .tourType(tourType)
-                    .startDate(addStartDate)
-                    .endDate(addEndDate)
-                    .pricePerUnit(addPricePerPerson)
-                    .hotelId(1) //stub
-                    .description(tourDescription)
-                    .discountId(1) //stub
-                    .build());
+            tourOffer.setTourType(tourType);
```

Comment:
Why not to use builder?

Reply:
Because we already have an object, we used to change only few fields, it’s unnecessary to build new object.